src/main/java/com/google/cloud/tools/appengine/api/devserver/RunConfiguration.java

```diff
@@ -53,6 +54,13 @@
 
   List<String> getJvmFlags();
 
+  /**
+   * Return an immutable map of environment variable strings.
+   * 
+   * @return the environment variables
+   */
+  ImmutableMap<String, String> getEnvironmentVariables();
```

Comment:
No, it shouldn't. An ImmutableMap is not a Map and cannot be used as one since it doesn't implement the write operations. This is the violation of the Liskov substitution principal I mentioned above. IMHO this is a design flaw in the Java Collections API, but it is not one we can fix here.